src/jit/lir.cpp

```diff
@@ -972,11 +972,19 @@ void LIR::Range::InsertAtEnd(Range&& range)
 // Arguments:
 //    node - The node to remove. Must be part of this range.
 //
-void LIR::Range::Remove(GenTree* node)
+void LIR::Range::Remove(GenTree* node, bool markOperandsUnused)
 {
     assert(node != nullptr);
     assert(Contains(node));
 
+    if (markOperandsUnused)
+    {
+        for (GenTree* operand : node->Operands())
+        {
+            operand->gtLIRFlags |= Flags::IsUnusedValue;
+        }
+    }
```

Comment:
I think it would be useful to have methods to set and clear the unused value flag, e.g. `LIR::Flags:SetUnusedValue()` and `LIR::Flags:ClearUnusedValue()` or something like that. It makes the intent clearer than the and-ing and or-ing.